src/rocketData.py

```diff
@@ -1,220 +1,66 @@
+    all_rocketdata
+        Description: returns an array of all sensor data to be used in convert_to_csv_string
 
-    How to call a setter and getter:
-    rocketdata = RocketData()
-    rocketdata.time_stamp <- this gets the valus of time stamp
-    rocketdata.time_stamp = 6.0 <- this Sets the value of time stamp to 6.0
+    convert_to_csv_string
+        Description: converts all of the data into a csv string
 
-    multi variable sets and set through lists:
+    send_to_airbrakes
+        Description: sends an object that contains altitude, velocity, and acceleration
+        info to calculations, which will send it to airbrakes (idk if this is how it works
+        but send_to_airbrakes will send an object with relevant info wherever it has to go)
 
-    rocketdata.imu = [5.0,6.0,7.0] <- sets in order of temperature, humidity, pressure see definitions of setters for more info
+    send_to_blackbox
+        Description: converts data to a csv string to be sent to blackbox
 
-    for setting all data use rocketdata = [imu_t, imu_h, imu_p, bme_y, bme_p, bme_r, bme_a, bme_v, sg_1, sg_2, sg_3, sg_4, sg_5, sg_6, sg_7, sg_8, sg_9, sg_10, sg_11, sg_12, e, ts]
+    send_to_ground
+        Description: takes critical data (what exactly?) and converts it to a string to be sent to the groundstation
+
+    """
+    data = {}
 
-    def data_dict_set(self, rd)
-        Description:
-            updates the self.data with a dictionary. Differnet from setters
-            because it does it updates the whole object not just individual
-            attributes
-        Parm:
-            rd: updated data object.
+    _lsm = None
+    _bme = None
+    _adx = None
+    _encoders = None
 
-    data_to_json()
-        Desciption:
-            Converts the dictionary to a Json format
+    def __init__(self):
+        self._bme = Bme()
+        self._lsm = Lsm()
+        self._adx = Adx()
+        self._encoders = Encoder() #there's no encoder class right now
 
-    print_jsondata()
-        Desciption:
-            prints the current state of the json to the terminal
+        self.data = {
+            # this might need to be restructured depending on how the sensor classes are done
+            'bme temperature': self._bme.read_temperature(),
+            'bme pressure': self._bme.read_pressure(),
+            'bme humidity': self._bme.read_humidity(),
+            'bme altitude': self._bme.read_altitude(),
 
-    all_rocketdata(self)
-        Description:
-            returns all data currently in self.data as an array
+            'lsm temperature': self._lsm.read_temperature(),
+            'lsm acceleration': self._lsm.read_acceleration(),
+            'lsm gyroscope': self._lsm.read_gyroscope(),
+            'lsm magnetometer': self._lsm.read_magnetometer(),
 
-    convert_to_csv_string(self):
-        Desciption:
-            returns a single csv string that contains all the data in self.data
+            'adx acceleration': self._adx.read_acceleration(),
```

Comment:
Maybe these should be nested?

not sure.

Like
```text
data {
  bme {
    temperature
    pressure
    humidity
    altitude
  }
  lsm {
    ...
  }
  adx {
    ...
  }
  strain_gauges [
    ...
  ]
}
```

Also for dictionary keys I think it looks better to have underscores instead of spaces, but not sure what we want for this project.

Reply:
Do you think it would be better to just have the sensors give us an object with their values/readings, like they did in the original code? If not, then yeah I think nesting them is probably better